content/client-lib-development-guide/features.textile

```diff
@@ -1336,6 +1339,32 @@ h4. ChannelProperties
 * @(CP2)@ The attributes of @ChannelProperties@ consist of:
 ** @(CP2a)@ @attachSerial@ string - contains the last @channelSerial@ received in an @ATTACHED@ @ProtocolMesage@ for the channel, see "RTL15a":#RTL15a
 
+h4. ChannelDetails
+
+* @(CHD1)@ @ChannelDetails@ is a type that represents information for a channel including channelID, name, status and occupancy 
+* @(CHD2)@ The attributes of @ChannelDetails@ consist of:
+** @(CHD2a)@ @channelID@ string - the identifier of the channel
+** @(CHD2b)@ @channelName@ string - the name of the channel
+** @(CHD2c)@ @status@ @ChannelStatus@ - status an object containing channel status information 
+
+h4. ChannelStatus
+
+* @(CHS1)@ @ChannelStatus@ is a type that contains status and occupancy for a channel 
+* @(CHS2)@ The attributes of @ChannelStatus@ consist of:
+** @(CHS2a)@ @isActive@ boolean - represents if the channel is active
+** @(CHS2b)@ @occupancy@ @ChannelMetrics@ - occupancy is an object containing the metrics for the channel
```

Comment:
`occupancy` is actually an object with a `metrics` field, and the value of `metrics` is a `ChannelMetrics` object:

```json
{
        "name": "example",
        "channelId": "example",
        "status": {
                "isActive": true,
                "occupancy": {
                        "metrics": {
                                "connections": 1,
                                "publishers": 1,
                                "subscribers": 1,
                                "presenceConnections": 1,
                                "presenceMembers": 0,
                                "presenceSubscribers": 1
                        }
                }
        }
}
```

So I think we need an extra object in between.

Reply:
Also, it concerns me that we are basically duplicating what is on the following docs page:

https://ably.com/docs/realtime/channel-metadata#types

the source of which is already contained within this same GitHub repository:

https://github.com/ably/docs/blob/main/content/partials/types/_channel_details.textile